components/Dialogs/src/MDCDialogPresentationController.m

```diff
@@ -330,9 +341,56 @@ - (void)keyboardWatcherHandler:(NSNotification *)aNotification {
                    animations:^{
                      CGRect presentedViewFrame = [self frameOfPresentedViewInContainerView];
                      self.presentedView.frame = presentedViewFrame;
-                     self.trackingView.frame = presentedViewFrame;
                    }
                    completion:NULL];
 }
 
+#pragma mark - MDMTransitionWithCustomDuration
+
+- (NSTimeInterval)transitionDurationWithContext:(id<MDMTransitionContext>)context {
+  if (context.direction == MDMTransitionDirectionForward) {
+    return MDCDialogTransitionMotionSpec.appearance.scrimOpacity.duration;
```

Comment:
Does the "scrimOpacity" just happen to be the longest timing in the transition animation or is there another reason why we use it to determine the overall duration? For example, if the scrim opacity animation is later delayed and shortened in duration, will that cause problems?

Reply:
Ahhh this is a super good question. I'm still not certain what the safest way to calculate "duration" for a transition is.

That being said, this duration API simply informs UIKit of the expected duration - the "true" duration is measured by the completion handlers of our animations. UIKit simply uses this expected duration to animate its own views (e.g. a navigation bar animating its transition) in tandem with our own.

I don't like the way this current code is written either, so I may just change the spec to include a transition duration property.

Reply:
Done.